articles/android-viewpager2/index.md

```diff
@@ -0,0 +1,214 @@
+val pager = findViewById<ViewPager2>(R.id.pager)
+```
+
+Then add the adapter as shown.
+
+```Kotlin
+pager.adapter = PagerAdapter(this, words)
+```
+
+That's what we need for the `ViewPager2`. Now build and run the application.
+
+The output should be like to this.
+
+![View pager gif](/engineering-education/android-viewpager2/view-pager.gif)
+
+To change the scroll orientation to vertical, add the following statement.
+
+```Kotlin
+pager.orientation = ViewPager2.ORIENTATION_VERTICAL
+```
+
+It should resemble the demo below.
+
+![vertical orientation gif](/engineering-education/android-viewpager2/vertical-scroll.gif)
+
+### Step 5 — Integrating TabLayout
+Now that our view pager is functioning as expected. Let's go ahead and integrate it with `TabLayout`.
+
+First add the reference to the `TabLayout` component.
+
+```Kotlin
+val tabLayout = findViewById<TabLayout>(R.id.tabLayout)
+```
+
+To integrate `ViewPager2` with `TabLayout` we need to use a `TabLayoutMediator` class. This was quite easier with the old ViewPager. One had to use the `TabLayout`'s `setUpWithViewPager` method and pass the `ViewPager2` refference.
+
+The `TabLayoutMediator` class takes in two parameters, the `TabLayout` and `ViewPager2` references.
+
+```Kotlin
+TabLayoutMediator(tabLayout, pager)
+```
+
+The class then takes a lambda function with two parameters. `TabLayout.Tab` and an integer representing the position the ViewPagers' pages.
+
+```Kotlin
+TabLayoutMediator(tabLayout, pager) {tab, position ->
+      tab.text = "${position + 1}"
+}.attach()
+```
+
+In the function, we set the tabs' text to the position plus one. We add one because the positions start from zero. We attach the components by calling the `attach` method.
+
+Build and run the application. This is how it should look.
+
+![tab layout gif](/engineering-education/android-viewpager2/tab-layout.gif)
+
+Another important feature of `TabLayout` is the on tab selected listener. This notifies the listeners whenever a tab's selected, unselected or reselected. This is useful when one wants to perform some background tasks when the listener fires up. We will use toasts in our application for demonstration.
+
+Add the following code to implement the feature.
+
+```Kotlin
+tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
+    override fun onTabSelected(tab: TabLayout.Tab?) {
+        Toast.makeText(this@MainActivity, "Tab ${tab?.text} selected", Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onTabUnselected(tab: TabLayout.Tab?) {
+        Toast.makeText(this@MainActivity, "Tab ${tab?.text} unselected", Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onTabReselected(tab: TabLayout.Tab?) {
+        Toast.makeText(this@MainActivity, "Tab ${tab?.text} reselected", Toast.LENGTH_SHORT).show()
+    }
+})
+```
+
+Build and run the app again.
+
+![toasts gif](/engineering-education/android-viewpager2/toasts.gif)
+
+### Conclusion
```

Comment:
Could you make the conclusion a bit longer. Maybe provide a summary of why to use ViewPager instead of other options or a summary of its pros.

Reply:
@LinusMuema - I have made the requested changes. Please review.